backend/routes/friends.js

```diff
@@ -88,11 +88,63 @@ router.get('/:userId', verifyToken, async (req, res) => {
         res.status(500).json({ error: 'Internal server error'})
     }
 });
+
 // friend recommendation
 router.get('/recommendedFriends/:userId', verifyToken, async (req, res) => {
     try{
         const userId = req.params.userId;
-        // fetch all post with their likes and comments
+
+          // fetch current user's friends - 1st degree
+          const friends = await Friend.findAll({ where: { userId }});
+          const friendIds = friends.map(friend => friend.friendId);
+
+          // helper function to fetch friends of  a given userId  excluding the current friends and the user
+          const excludeFriends = async (userIds, excludeIds) => {
+            return await Friend.findAll({
+                where: {
+                    userId: { [Op.in]: userIds},
+                    friendId: { [Op.notIn]: excludeIds}
+                }
+            });
+          };
+
+          // friends of friends - 2nd degree
+          const friendsOfFriends = await excludeFriends(friendIds, [...friendIds, userId]);
+
+
+          // count the occurrences of each friendId for friend of friends (2nd degree)
+          const countFriends = friendsOfFriends.reduce((acc, friend) => {
+              acc[friend.friendId] = (acc[friend.friendId] || 0) + 1;
+              return acc;
+          }, {});
+
+          // fetch friends of friends of friends 3rd degree
+          const secondDegreeFriendIds = Object.keys(countFriends).map(id => parseInt(id));
```

Comment:
based on the naming of `countFriends` i would have expected it to be a number... what is its type?